src/helpers/http/mod.rs

```diff
@@ -163,6 +205,74 @@ mod e2e_tests {
         helper
     }
 
+    #[tokio::test]
+    async fn prss_key_exchange() {
+        logging::setup();
+
+        let peer_discovery = peer_discovery();
+        let h1 = init_helper(Role::H1, &peer_discovery).await;
+        let h2 = init_helper(Role::H2, &peer_discovery).await;
+        let h3 = init_helper(Role::H3, &peer_discovery).await;
+
+        let gateway1 = h1.query(QueryId).unwrap();
+        let gateway2 = h2.query(QueryId).unwrap();
+        let gateway3 = h3.query(QueryId).unwrap();
+
+        let step = Step::default().narrow(&PrssExchangeStep);
+        let mut rng1 = StdRng::from_entropy();
+        let mut rng2 = StdRng::from_entropy();
+        let mut rng3 = StdRng::from_entropy();
+
+        let participant1 = h1.prss_endpoint(&gateway1, &step, &mut rng1);
+        let participant2 = h2.prss_endpoint(&gateway2, &step, &mut rng2);
+        let participant3 = h3.prss_endpoint(&gateway3, &step, &mut rng3);
+        let (participant1, participant2, participant3) =
+            tokio::try_join!(participant1, participant2, participant3).unwrap();
+
+        let rbg = RandomBitsGenerator::<Fp31>::new();
+        let ctx1 = h1.context(&gateway1, &participant1, &rbg);
+        let ctx2 = h2.context(&gateway2, &participant2, &rbg);
+        let ctx3 = h3.context(&gateway3, &participant3, &rbg);
+
+        let idx = 0u128;
+        let (left1, right1) = ctx1.prss().generate_values(idx);
+        let (left2, right2) = ctx2.prss().generate_values(idx);
+        let (left3, right3) = ctx3.prss().generate_values(idx);
+
+        assert_eq!(left1, right3);
+        assert_eq!(right1, left2);
+        assert_eq!(right2, left3);
+
+        // recreate participants and ensure values are different
+        let step = step.narrow("second_time");
+        let participant1 = h1.prss_endpoint(&gateway1, &step, &mut rng1);
```

Comment:
I think this is fine because rng is used to generate 32 byte value. It is VERY unlikely that it generates the same value twice. Alternatively, `prss_endpoint` can take a seed (for prod seed should be generated using CPRNG) and initialize a random generator from it. That would make this test deterministic if you provide different seeds. Don't think it is a big deal here, but in general don't be afraid to change the API to make it testable more easily